### PR TITLE
[trino][compute] Add compute support for Trino

### PR DESCRIPTION
**Pull request by amitsrivastava**
With this change, trino compute can be discovered, saved to db and used similar to hive and impala.

### CODE REVIEW

File: apps/beeswax/src/beeswax/common.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
 
 RELATION_OPS = ['=', '<>', '<', '<=', '>', '>='] + RELATION_OPS_UNARY
 
-COMPUTE_TYPES = ['hive-compute', 'impala-compute']
+COMPUTE_TYPES = ['hive-compute', 'impala-compute', 'trino-compute', 'trino']
```
Comment (Harshg999) [inline apps/beeswax/src/beeswax/common.py:47]: Why are we adding both `trino-compute` and `trino`?

File: desktop/core/src/desktop/management/commands/sync_warehouses.py [hunk 2]
```diff
@@ -224,3 +234,44 @@ def update_impala_configs(namespace, impala, host):
...
+    settings.append({"name": "auth_password_script", "value": "/etc/hue/conf/altscript.sh hue.binduser.password"})
+    if trino_coordinator_ingress and trino_coordinator_ingress.spec.rules:
+      coordinator_url = 'https://%s:443' % trino_coordinator_ingress.spec.rules[0].host
+
+  settings.append({"name": "url", "value": coordinator_url})
+
+  trino.update({
+    'dialect': 'trino',
+    'interface': 'trino',
+    # 'ldap_groups': ldap_groups.split(",") if ldap_groups else None,
```
Comment (Harshg999) [inline desktop/core/src/desktop/management/commands/sync_warehouses.py:275]: Nit: To remove?

File: apps/beeswax/src/beeswax/common.py [hunk 3]
```diff
@@ -135,7 +143,9 @@ def find_compute(cluster=None, user=None, dialect=None, namespace_id=None):
       dialect = selected_compute['dialect'] if selected_compute.get('dialect') else dialect
       if not dialect and cluster.get('type'):
         t = cluster['type']
-        dialect = 'hive' if t.startswith('hive') else 'impala' if t.startswith('impala') else None
+        dialect = 'hive' if t.startswith('hive') else\
+          'impala' if t.startswith('impala') else\
+            'trino' if t.startswith('trino') else None
```
Comment (Harshg999) [inline apps/beeswax/src/beeswax/common.py:149]: Nit: Is formatting correct?

File: desktop/core/src/desktop/models.py [hunk 4]
```diff
@@ -1909,7 +1909,7 @@ def _get_editor(self):
           'optimizer': get_optimizer_mode(),
           'page': '/editor/?type=%(type)s' % interpreter,
           'is_sql': interpreter['is_sql'],
-          'is_batchable': interpreter['dialect'] in ['hive', 'impala'] or interpreter['interface'] in ['oozie', 'sqlalchemy'],
+          'is_batchable': interpreter['dialect'] in ['hive', 'impala', 'trino'] or interpreter['interface'] in ['oozie', 'sqlalchemy'],
```
Comment (Harshg999) [inline desktop/core/src/desktop/models.py:1912]: Is this required to enable batch queries for Trino?

**Review by Harshg999 — COMMENTED** [review]
I've added few review comments. Also possible to add unit tests for this change?

File: desktop/core/src/desktop/models.py (hunk 4 above)
Comment (amitsrivastava) [reply desktop/core/src/desktop/models.py:1912]: If hive and impala are batch-able then why not trino? it seems to work fine on the frontend. Is there a reason not to do it?

File: apps/beeswax/src/beeswax/common.py (hunk 3 above)
Comment (amitsrivastava) [reply apps/beeswax/src/beeswax/common.py:149]: pycharm likes it this way and ruff also doesn't complain so I didn't bother much. On a closer look, the L148 is actually part of the `if` started on L147, so an extra indent might be okay.

File: desktop/core/src/desktop/management/commands/sync_warehouses.py (hunk 2 above)
Comment (amitsrivastava) [reply desktop/core/src/desktop/management/commands/sync_warehouses.py]: done.

File: apps/beeswax/src/beeswax/common.py (hunk 1 above)
Comment (amitsrivastava) [reply apps/beeswax/src/beeswax/common.py]: cleaned up. thanks.

**Review by Harshg999 — APPROVED** [review]
LGTM! Few last comments..

File: desktop/libs/notebook/src/notebook/connectors/trino.py [hunk 5]
```diff
@@ -47,7 +45,7 @@ def decorator(*args, **kwargs):
     except RestException as e:
       try:
         message = force_unicode(json.loads(e.message)['errors'])
-      except:
+      except BaseException as t:
```
[the remark this answers is not in the document]
Comment (amitsrivastava) [reply desktop/libs/notebook/src/notebook/connectors/trino.py:48]: `ruff` doesn't like plain `except:` so I was forced to add the rest. What's the recommended way to handle all kinds of exception in py3?

**Review by athithyaaselvam — APPROVED** [review]
LGTM!

**Pull request merged**